test/battle/ability/stalwart.c

```diff
@@ -0,0 +1,55 @@
+#include "global.h"
+#include "test/battle.h"
+
+DOUBLE_BATTLE_TEST("Stalwart ignores redirection from Follow-Me")
+{
+    GIVEN {
+        PLAYER(SPECIES_ARCHALUDON) { Ability(ABILITY_STALWART); }
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponentLeft, MOVE_FOLLOW_ME); MOVE(playerLeft, MOVE_DRACO_METEOR, target: opponentRight); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_FOLLOW_ME, opponentLeft);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_DRACO_METEOR, playerLeft);
+        HP_BAR(opponentRight);
+        NOT HP_BAR(opponentLeft);
+    }
+}
+
+DOUBLE_BATTLE_TEST("Stalwart stops Lightning Rod and Storm Drain from redirecting moves")
+{
+    u32 ability;
+    PARAMETRIZE { ability = ABILITY_STORM_DRAIN; }
+    PARAMETRIZE { ability = ABILITY_LIGHTNING_ROD; }
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_SPARK].type == TYPE_ELECTRIC);
+        ASSUME(gMovesInfo[MOVE_WATER_GUN].type == TYPE_WATER);
+        PLAYER(SPECIES_WOBBUFFET) { Ability(ABILITY_STALWART); }
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_RAICHU) { Ability(ability); }
```

Comment:
could you parametrize the species here as well so storm drain is attached to a legal ability?